plugins/velocity/src/main/java/org/apache/struts2/views/velocity/VelocityManager.java

```diff
@@ -161,32 +164,24 @@ public Context createContext(ValueStack stack, HttpServletRequest req, HttpServl
      * perform any initialization of the contexts.  All that must be done in the
      * context itself.
      *
-     * @param servletRequest the servlet request object
+     * @param servletRequest  the servlet request object
      * @param servletResponse the servlet response object
-     * @param extraContext map with extra context
+     * @param extraContext    map with extra context
      * @return an VelocityContext[] of contexts to chain
      */
-    protected VelocityContext[] prepareChainedContexts(HttpServletRequest servletRequest, HttpServletResponse servletResponse, Map extraContext) {
-        if (this.chainedContextNames == null) {
-            return null;
-        }
-        List contextList = new ArrayList();
-        for (int i = 0; i < chainedContextNames.length; i++) {
-            String className = chainedContextNames[i];
-            try {
-                VelocityContext velocityContext = (VelocityContext) objectFactory.buildBean(className, null);
-                contextList.add(velocityContext);
-            } catch (Exception e) {
-                LOG.warn("Warning. {} caught while attempting to instantiate a chained VelocityContext, {} -- skipping", e.getClass().getName(), className);
+    protected List<VelocityContext> prepareChainedContexts(HttpServletRequest servletRequest, HttpServletResponse servletResponse, Map<String, Object> extraContext) {
+        List<VelocityContext> contextList = new ArrayList<>();
+        if (this.chainedContextNames != null) {
+            for (String className : chainedContextNames) {
+                try {
+                    VelocityContext velocityContext = (VelocityContext) objectFactory.buildBean(className, extraContext);
+                    contextList.add(velocityContext);
+                } catch (Exception e) {
+                    LOG.warn("Warning. {} caught while attempting to instantiate a chained VelocityContext, {} -- skipping", e.getClass().getName(), className);
+                }
             }
         }
-        if (contextList.size() > 0) {
-            VelocityContext[] extraContexts = new VelocityContext[contextList.size()];
-            contextList.toArray(extraContexts);
-            return extraContexts;
-        } else {
-            return null;
-        }
+        return contextList;
```

Comment:
Possible (but _unlikely_) that someone extended the class and was expecting the null to detect empty `contextList` or `this.chainedContextNames`.

Reply:
I changed JavaDoc to reflect the new behaviour